lib/app/routes/problems.rb

```diff
@@ -27,6 +28,47 @@ class Problems < Core
         problem.validate or halt 404, { error: problem.error }.to_json
         pg :problem_test, locals: { problem: problem }
       end
+
+      get '/problems' do
+        all_languages = Xapi::Config.languages # Xapi::Config.all
+        
+
+        slugs={}
+
+
+        all_languages.each do |language|
+          track = Xapi::Config.find(language)
+          problems = track.problems
+          problems.each do |problem|
+            if slugs[problem.name].nil?
+              slugs[problem.name] = {}
+              slugs[problem.name]["languages"] = []
+              slugs[problem.name]["name"] = problem.name
+              slugs[problem.name]["slug"] = problem.slug
+              theYaml = "./metadata/#{problem.slug}.yml"
```

Comment:
Ruby typically uses snake case for variables. How about `yaml_file` instead of `theYaml`?